jerry-core/vm/vm.c

```diff
@@ -1651,6 +1657,36 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         }
         case VM_OC_ADD:
         {
+          if (ecma_are_values_integer_numbers (left_value, right_value))
```

Comment:
It seems the helper function requires several parameters, so it does not look really nice. Macros make hard to debug things.

The VM is the most critical element of the engine performance, so we use techniques here which is not used in the other parts of Jerry. E.g. gotos, and code duplications. I know this increases the maintenance burden (and binary size), but the gain here outweights the costs. For binary size optimizations, we usually focus on rarely used but large parts of the code, e.g. json support.